inference-engine/src/mkldnn_plugin/nodes/mkldnn_shuffle_channels_node.h

```diff
@@ -18,23 +18,43 @@ class MKLDNNShuffleChannelsNode : public MKLDNNNode {
     MKLDNNShuffleChannelsNode(const std::shared_ptr<ngraph::Node>& op, const mkldnn::engine& eng, MKLDNNWeightsSharing::Ptr &cache);
     ~MKLDNNShuffleChannelsNode() override = default;
 
-    void getSupportedDescriptors() override;
+    static bool isSupportedOperation(const std::shared_ptr<const ngraph::Node>& op, std::string& errorMessage) noexcept;
+    void getSupportedDescriptors() override {};
     void initSupportedPrimitiveDescriptors() override;
     void createPrimitive() override;
     void execute(mkldnn::stream strm) override;
     bool created() const override;
 
-    static bool isSupportedOperation(const std::shared_ptr<const ngraph::Node>& op, std::string& errorMessage) noexcept;
+    void prepareParams() override;
+
+protected:
+    void executeDynamicImpl(mkldnn::stream strm) override { execute(strm); };
 
 private:
-    ngraph::Shape inShape_;
-    int dataRank_;
-    int axis_;
-    size_t group_;
-    size_t groupSize_;
-
-    std::unique_ptr<PermuteKernel> permuteKernel_;
-    bool supportDynamicBatch_;
+    struct ShuffleChannelsAttributes {
+        LayoutType layoutType;
```

Comment:
the same as for `Normalize`
please extract all filed which can be changed in inference time and pass them as `ShuffleChannelsExecutor` ctor arguments

Reply:
Yeah, sure. Done, thanks